java/test/src/main/java/org/ray/api/test/ActorConcurrentCallsTest.java

```diff
@@ -0,0 +1,107 @@
+package org.ray.api.test;
+
+import com.google.common.collect.ImmutableList;
+import org.ray.api.Ray;
+import org.ray.api.RayActor;
+import org.ray.api.RayObject;
+import org.ray.api.TestUtils;
+import org.ray.api.annotation.RayRemote;
+import org.ray.api.options.ActorCreationOptions;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+
+
+@Test(groups = {"directCall"})
+public class ActorConcurrentCallsTest extends BaseTest {
+
+  @RayRemote
+  public static class ConcurrentActor {
+
+    private BlockingQueue<Integer> queue = new LinkedBlockingQueue<>(3);
+
+    public ConcurrentActor() {
+
+    }
+
+    public List<Integer> append(int x) {
+      try {
+        queue.put(x);
+      } catch (InterruptedException e) {
+        throw new RuntimeException(e);
+      }
+
+      if (queue.size() == 3) {
+        queue.notifyAll();
+      } else {
+        try {
+          queue.wait();
+        } catch (InterruptedException e) {
+          throw new RuntimeException(e);
+        }
+      }
+      List<Integer> li = new ArrayList<>(3);
+      li.addAll(queue);
+      li.sort(new Comparator<Integer>() {
+        public int compare(Integer o1, Integer o2) {
+          return o1.compareTo(o2);
+        }
+      });
+      return li;
+    }
+
+  }
+
+  public void testConcurrentCalls() {
```

Comment:
could you add some comments on what this is testing for?

Reply:
and it might make sense to add some test to verify the methods are indeed executed in parallel.

Reply:
It's actually that the case must executed in parallel because the first task is pending until other tasks putting items into the queue.